scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2074,58 +2086,72 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       val op = termName()
       if (op.value == "-" && token.is[NumericLiteral])
         simpleExprRest(atPos(op, auto)(literal(isNegated = true)), canApply = true)
-      else
-        atPos(op, auto)(Term.ApplyUnary(op, simpleExpr(allowRepeated = true)))
+      else {
+        simpleExpr0(allowRepeated = true) match {
+          case Right(result) => atPos(op, auto)(Term.ApplyUnary(op, result))
+          case Left(error) =>
+            // maybe it is not unary operator but simply an ident `trait - {...}`
+            // we would fail here anyway, let's try to treat it as ident
+            simpleExprRest(op, canApply = true)
+        }
+      }
     }
 
   def simpleExpr(allowRepeated: Boolean): Term = autoPos {
+    simpleExpr0(allowRepeated) match {
+      case Right(result) => result
+      case Left(error) => throw error
```

Comment:
Maybe let's add a test case with this failing?

Reply:
added in test "simple-expression-parse-error"

Reply:
if you had used `Try`, this would be just `.get`